admin-scripts/extract_mini_repository.sh

```diff
@@ -0,0 +1,118 @@
+#! /bin/sh
+
+set -e
+
+if [ $# = 0 ]; then
+   cat <<EOF
+
+This script is able to generate a minimalistic and self-contained 
+opam-repository for a given set of packages and compiler versions.
+The resulting repository will be in './opam-mini-repository.tar.gz'
+
+Example:
+
+ ## Including only the latest compiler version:
+ $0 ocp-indent lwt
+
+ ## Including a specific verion of OCaml or multiple version.
+ $0 ocaml.4.02.1 ocaml.4.01.0 ocp-indent lwt
+
+EOF
+   exit 1
+fi
+
+DEFAULT_COMPILER=4.02.1
+OPAM_REPO=https://github.com/ocaml/opam-repository/archive/master.tar.gz
+
+TARGET=$(pwd)
+DIR=$(mktemp -d)
```

Comment:
need a `set -e` somewhere, or if this fails then the `rm -rf` below may well delete `/`